### ISSUE TITLE
Bad substitution issues on shell script

### ISSUE THREAD

**Issue by mallorybowes** [mallorybowes/chrome-mal-ids#10]
Yep, I saw it.  I'll do some more research tomorrow and fix it but the script will still check for matching malicious extensions ok.  (The function that has the issue has to do with the downloaded file integrity check.)

![image](https://user-images.githubusercontent.com/24720141/89375907-54991f80-d6b4-11ea-88f2-46096aef2bd2.png)

**Comment by mallorybowes** [comment]
I confirmed the code worked before merging the pull request and I've done the following to rule out issues:

- **Passed**: Re-committed newly concatenated extension list and checksum file.  (Testing for bad / hidden characters in source extension files.  I also initially forgot the new chksum file but everything in the repo now matches.  I'm thinking I should get into the realm of a build script to keep me from making the tired, blurry-eyed mistakes again.  :-) )
- **Passed**:  Tested new extension list with old script logic (looping through files instead of shell expansion) and the old script with new extension list file worked as expected.  Wanted to make sure it wasn't an input file issue.
- **Passed**:  Check for recent updates to bash shell on testing machine.  Looks like the current version I tested with (4.2.20) was updated last in Aug. 2019.
- **Passed**:  Ran current script through shellcheck and incorporated suggested changes.  Still produced the same errors.  I did not commit that version of the script to the repo.
- **Failed**:  Ran current script using local copies of multiple extension list files and local chksum files (removed downloading variable).  Error was still produced using old 251 line extension list, separate file for the new 295 extensions, and concatenated 546 line file with matching chksums.
- **Failed**:  Tested script on second Linux machine.  Still produced errors.

One thing that does look weird in the debug is for the following array comparison:
![image](https://user-images.githubusercontent.com/24720141/89418923-58e62c80-d6f6-11ea-9d1f-24700c450353.png)
Looking for those extension names before that operator does not exist in any of the source extension files.  Extensions after the =~ operator, the extension IDs do exist in the source files.
![image](https://user-images.githubusercontent.com/24720141/89417955-022c2300-d6f5-11ea-81c3-837153fbf222.png)
![image](https://user-images.githubusercontent.com/24720141/89417900-f04a8000-d6f4-11ea-985e-bb0eb30768cd.png)

Going to ask the contrib of that portion of the script to see if he can see what I'm missing.  (I'm a hack, not a programmer, as anyone can tell by looking at my contribs...  :-) )
